### PR TITLE
Add HLD for SONiC Debian Upgrade Cadence

### PR DESCRIPTION
**Pull request by pavannaregundi**
HLD to document Debian upgrade cadence in SONiC.

### CODE REVIEW

**Comment by pavannaregundi** [comment]
@saiarcot895 please check.

File: doc/debian_upgrade/SONiC_Debian_Upgrade_Cadence.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+	* Linux Kernel
+* Target timelines for Linux Kernel minor version upgrades on different SONiC branches.
+	* Branches based on Debian 'Stable' - 6 months.
+	* Branches based on Debian 'OldStable' - 6 months.
+	* Branches Based on Debian 'OldOldStable' - Based on requirement.
+		* Ex: As per the current status below is the mapping of SONiC branches to Debian releases.
+			* OldOldStable (Buster) -  202006, 202012, 202106
+			* OldStable (Bullseye) - 202111, 202205, 202211, 202305
+			* Stable (Bookworm) - 202311, master
+* Other third-party list below will be updated based on requirement.
```
Comment (saiarcot895) [inline doc/debian_upgrade/SONiC_Debian_Upgrade_Cadence.md:171]: Can we add `dockerd` to this list as well?

Comment (pavannaregundi) [reply doc/debian_upgrade/SONiC_Debian_Upgrade_Cadence.md:171]: done.

**Comment by zhangyanzhao** [comment]
community review recording https://zoom.us/rec/share/pbptYwEUFOgPuNuQc-dBqxYsUfIO9Ezh4EZ8VlHWZ8wHQIC-IDJyanGvZ84YltDb.OrFYH4lQQ7-QbJLp

**Comment by bluecmd** [comment]
FWIW, there are some work that was done by me last year in https://github.com/sonic-net/sonic-buildimage/pull/17075. Feel free to use that work in any way you seem fit.

**Review by zhangyanzhao — APPROVED** [review]

**Pull request merged**